tests/unit/mathematics/linalg/operations/Eigen3_operations_unittest.cc

```diff
@@ -40,26 +50,10 @@ class LinalgBackendEigenNonIntegerTypesTest : public ::testing::Test
 {
 };
 
-// TODO: make global definitions
-// Definition of the 4 groups of Shogun types
-// (shogun/mathematics/linalg/LinalgBackendBase.h)
-// TODO: add bool and complex128_t types
-typedef ::testing::Types<
-    int8_t, int16_t, uint16_t, int32_t, uint32_t, int64_t, uint64_t, float32_t,
-    float64_t, floatmax_t, char>
-    AllTypes;
-typedef ::testing::Types<
-    int8_t, int16_t, uint16_t, int32_t, uint32_t, int64_t, uint64_t, float32_t,
-    float64_t, floatmax_t>
-    NonComplexTypes;
-typedef ::testing::Types<float32_t, float64_t, floatmax_t> RealTypes;
-// TODO: add complex128_t type
-typedef ::testing::Types<float32_t, float64_t, floatmax_t> NonIntegerTypes;
-
-TYPED_TEST_CASE(LinalgBackendEigenAllTypesTest, AllTypes);
-TYPED_TEST_CASE(LinalgBackendEigenNonComplexTypesTest, NonComplexTypes);
-TYPED_TEST_CASE(LinalgBackendEigenRealTypesTest, RealTypes);
-TYPED_TEST_CASE(LinalgBackendEigenNonIntegerTypesTest, NonIntegerTypes);
+TYPED_TEST_CASE(LinalgBackendEigenAllTypesTest, TypesGoogleTestWrapper<sg_all_primitive_types>::type);
```

Comment:
this is sooo nice now!
should have an entrance task to do more of this!

Reply:
I also wrote some structs to pop types from those structs. For example, you want sg_all_primitive_types, but you don't want to test for complex128_t, then you could do something like `popTypesGoogleTestWrapper<sg_all_primitive_type, complex128_t>::type`. Would that be useful to add? Either that or inside the test we can write type checks but that could be repetitive...

Reply:
yes definitely!

Reply:
The only thing is that I need to implement algorithms in compile time for that (append, reverse and pop). So it's a bit of a monster code, but it gets the job done and I have tests for that...

Reply:
mmmh....not sure if it is worth it then?
As we can alway declare a new list in case it is required ?
How much code and how messy are we talking? :)

Reply:
Yes, we can declare a new one. The idea of just being able to remove some in the spot is that redefining these types might lead to someone forgetting one, mostly when there are more than 10 primitive types...
We are talking about 200 lines of code, probably less if I aim for conciseness.. In terms of mess it depends if you like reading recursive structs..

Reply:
ah actually less, I have a lot of things now that I don't need for this, so more like 100 loc